### PR TITLE
Add stroke path to Path2D interface per html canvas spec 

### PR DESCRIPTION
**Pull request by cleemesser**
* add stroke(path: Path2D) form to canvas,, see issue #334 
current deficiencies I am aware off: 
- [ ] lacks tests
- [ ] lacks documentation

### CODE REVIEW

**Comment by martinRenou** [comment]
Looks great!

> lacks tests

This could be as simple as adding a new cell at the end of this Notebook testing the feature:
https://github.com/jupyter-widgets-contrib/ipycanvas/blob/master/ui-tests/tests/notebooks/ipycanvas.ipynb

**Comment by martinRenou** [comment]
CI failures look unrelated, I will fix them in parallel

**Comment by martinRenou** [comment]
Done and rebased. There is a linting failure that comes from this PR.

**Comment by cleemesser** [comment]
> Looks great!
> 
> > lacks tests
> 
> This could be as simple as adding a new cell at the end of this Notebook testing the feature: https://github.com/jupyter-widgets-contrib/ipycanvas/blob/master/ui-tests/tests/notebooks/ipycanvas.ipynb

- I've prepared a simple test for ui-tests/notebooks/ipycanvas.ipynb
- I tried black --check on my repo files and don't get any changes. not sure what needs to be fixed from linking/formatting point of view. maybe look again soon

**Comment by martinRenou** [comment]
Thanks!

> I tried black --check on my repo files and don't get any changes

`black --check` only checks without making changes (for testing the linting), you may want to run just `black` and commit the changes black makes.

**Comment by martinRenou** [comment]
bot please update snapshots

**Comment by cleemesser** [comment]
I went ahead and ran black again on canvas.py and pushed the commit.
BTW: I noticed that black wants  to a lots of reformats on the ui-test ipycanvas.ipynb notebook. I did not include those changes in my submission, but could in a future PR..

**Comment by martinRenou** [comment]
Thanks!

> I went ahead and ran black again on canvas.py and pushed the commit.

I don't see your new commit, are you sure you pushed?

> I did not include those changes in my submission, but could in a future PR

Sounds good!

**Comment by cleemesser** [comment]
Ok, started working on documentation. But seeing some unexpected behavior. 

when I extend the documentation example to include a 5th stroked path, the shape is not only stroked, it is *also* filled. I expected this to be a sinusoidal (non-filled) figure when rendered.

 Somehow, a fill is also being executed. 

This is what the cell looks like in ipycanvas.ipynb after my edit
```python
from ipycanvas import Canvas, Path2D

canvas = Canvas(width=350, height=350)

path1 = Path2D('M80 80 A 45 45, 0, 0, 0, 125 125 L 125 80 Z')
path2 = Path2D('M230 80 A 45 45, 0, 1, 0, 275 125 L 275 80 Z')
path3 = Path2D('M80 230 A 45 45, 0, 0, 1, 125 275 L 125 230 Z')
path4 = Path2D('M230 230 A 45 45, 0, 1, 1, 275 275 L 275 230 Z')

canvas.fill_style = 'green'
canvas.fill(path1)

canvas.fill_style = 'purple'
canvas.fill(path2)

canvas.fill_style = 'red'
canvas.fill(path3)

canvas.fill_style = 'blue'
canvas.fill(path4)

path5 = Path2D("M 10 150 Q 52.5 10, 95 150 T 180 150")
canvas.stroke_style = 'black'
canvas.line_width = 2
canvas.stroke(path5)
canvas
```

The equivalent in [javascript/html5 via jsfiddle]( https://jsfiddle.net/cleemesser/eLt52ybr/5/)
Shows the expected sinusoidal figure without filling.
code shown below:

```html
<canvas id="canvas" width=350 height=350></canvas>

```

```js
const canvas = document.getElementById("canvas");
const ctx = canvas.getContext("2d");

path1 = new Path2D('M80 80 A 45 45, 0, 0, 0, 125 125 L 125 80 Z')
path2 = new Path2D('M230 80 A 45 45, 0, 1, 0, 275 125 L 275 80 Z')
path3 = new Path2D('M80 230 A 45 45, 0, 0, 1, 125 275 L 125 230 Z')
path4 = new Path2D('M230 230 A 45 45, 0, 1, 1, 275 275 L 275 230 Z')

ctx.fillStyle = 'green'
ctx.fill(path1)

ctx.fillStyle = 'purple'
ctx.fill(path2)


ctx.fillStyle = 'red'
ctx.fill(path3)

ctx.fillStyle = 'blue'
ctx.fill(path4)

path5 = new Path2D("M 10 150 Q 52.5 10, 95 150 T 180 150")
ctx.strokeStyle='black'
ctx.lineWidth=2.0
ctx.stroke(path5)
```

**Comment by martinRenou** [comment]
bot please update snapshots

File: docs/drawing_paths.rst [hunk 1]
```diff
@@ -6,8 +6,8 @@ There are two ways for creating and drawing a path in ipycanvas.
 Using Path2D
 ------------
 
-You can define a Path2D given an SVG path. Note that once the path is created, it is read only, you cannot dynamically change the path value.
-Using the Path2D class is very useful and efficient when you want to reuse the same path multiple times.
+You can define a Path2D given an SVG path. Note that once the path is created, it is read only, you cannot dynamically change the path value. This means they do not (yet) support Path2D methods like Path2D.move_to, draw_line, etc.
```
Comment (martinRenou) [inline docs/drawing_paths.rst:9]:
```suggestion
You can define a Path2D given an SVG path. Note that once the path is created, it is read only, you cannot dynamically change the path value. ``ipycanvas`` does not (yet) support Path2D methods like Path2D.move_to, draw_line, etc.
```

**Review by martinRenou — CHANGES_REQUESTED** [review]
Small doc change suggestion, otherwise looks great thanks!

**Comment by martinRenou** [comment]
Triggering CI

**Review by martinRenou — APPROVED** [review]
Thanks!

**Pull request merged**